src/hostname/hostnamectl.c

```diff
@@ -315,23 +315,35 @@ static int show_all_names(sd_bus *bus) {
         return print_status_info(&info);
 }
 
+static int get_hostname_based_on_flag(sd_bus *bus) {
+        const char *attr;
+
+        if (!!arg_static + !!arg_pretty + !!arg_transient > 1)
+                return log_error_errno(SYNTHETIC_ERRNO(EINVAL),
+                                       "Cannot query more than one name type at a time");
+
+        attr = arg_pretty ? "PrettyHostname" :
+                arg_static ? "StaticHostname" : "Hostname";
+
+        return get_one_name(bus, attr, NULL);
+}
+
 static int show_status(int argc, char **argv, void *userdata) {
         sd_bus *bus = userdata;
 
-        if (arg_pretty || arg_static || arg_transient) {
-                const char *attr;
+        if (arg_pretty || arg_static || arg_transient)
+                return get_hostname_based_on_flag(bus);
 
-                if (!!arg_static + !!arg_pretty + !!arg_transient > 1)
-                        return log_error_errno(SYNTHETIC_ERRNO(EINVAL),
-                                               "Cannot query more than one name type at a time");
+        return show_all_names(bus);
+}
 
-                attr = arg_pretty ? "PrettyHostname" :
-                        arg_static ? "StaticHostname" : "Hostname";
+static int get_hostname(int argc, char **argv, void *userdata) {
+        sd_bus *bus = userdata;
 
-                return get_one_name(bus, attr, NULL);
-        }
+        if (arg_pretty || arg_static || arg_transient)
+                return get_hostname_based_on_flag(bus);
 
-        return show_all_names(bus);
+        return get_one_name(bus, "StaticHostname", NULL);
```

Comment:
i am pretty sure this should be the regular hostname, not the static one

Reply:
Yes, you're right. I see know that it can be even simplified just by calling `get_hostname_based_on_flag` directly, hence I do it in this way